### PR TITLE
Enable NavigationThreadingOptimizations by default in nightly

### PR DESCRIPTION
**Pull request by atuchin-m**
For https://github.com/brave/brave-browser/issues/23267

Update: https://github.com/brave/brave-browser/issues/23267#issuecomment-1154014387

### CODE REVIEW

**Review by iefremov — APPROVED** [review]

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers 👍

**Pull request merged**
